### PR TITLE
Edited ltp_fs.py, enhanced it to run when no external disk is given

### PR DESCRIPTION
**Pull request by Naresh-ibm**
if some one does not have external disk to pass it and wants to run
ltp, then this script fails, due to disk value as None.
So edited it and Now it will run on given/source directory without
mounting any disk on it and one can with external disk too..

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[lpt_fs_without_disk.tar.gz](https://github.com/avocado-framework-tests/avocado-misc-tests/files/844281/lpt_fs_without_disk.tar.gz)
[lpt_fs_with_disk.tar.gz](https://github.com/avocado-framework-tests/avocado-misc-tests/files/844282/lpt_fs_with_disk.tar.gz)

File: io/disk/ltp_fs.py [hunk 1]
```diff
@@ -42,22 +42,21 @@ def setUp(self):
...
-        part_obj = Partition(self.disk, mountpoint=self.mount_point)
-        self.log.info("Unmounting the disk or dir if it is already mounted")
-        part_obj.unmount()
-        self.log.info("creating %s file system on %s", self.fs, self.disk)
-        part_obj.mkfs(self.fs)
-        self.log.info("mounting %s on %s", self.disk, self.mount_point)
-        part_obj.mount()
+
+        if self.disk is not None:
+            self.part_obj = Partition(self.disk, mountpoint=self.mount_point)
```
Comment (nasastry) [inline io/disk/ltp_fs.py]: Local object creation is sufficient.  The same applies to the following.

File: io/disk/ltp_fs.py [hunk 2]
```diff
@@ -42,22 +42,21 @@ def setUp(self):
         for package in ['gcc', 'make', 'automake', 'autoconf']:
             if not sm.check_installed(package) and not sm.install(package):
                 self.error(package + ' is needed for the test to be run')
```
Comment (abdhaleegit) [inline io/disk/ltp_fs.py]: cab this be ?
self.error("%s is needed for the test to be run" % package)

File: io/disk/ltp_fs.py (hunk 1 above)
Comment (nasastry) [inline io/disk/ltp_fs.py]: Ignore my above comment. You are using self.part_obj in 'Teardown'.

**Comment by Naresh-ibm** [comment]
[Uploading lpt_fs_without_disk.tar.gz…]()
[Uploading lpt_fs_with_disk.tar.gz…]()

**Comment by Naresh-ibm** [comment]
Modified the changes suggested. please review and let me know if anything needed further to be modified.

**Comment by PraveenPenguin** [comment]
@abdhaleegit @nasastry thanks

**Pull request merged**